storage/backend_key.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"encoding/binary"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// backendKeyBytesLen is the length of backendKey bytes.
+// First a few bytes stores the revision of backendKey. The following byte
+// is a ' ' as delimiter. The last 4 bytes is the event type in big-endian
+// format.
+const backendKeyBytesLen = revBytesLen + 1 + 4
+
+type backendKey struct {
+	rev revision
+	t   storagepb.Event_EventType
```

Comment:
why do we really need the event type anymore? The point of this is to decouple the actual storage layer and the event notification layer. The KV does not need to have any knowledge about watchableKV. The event is a concept from upper layer.